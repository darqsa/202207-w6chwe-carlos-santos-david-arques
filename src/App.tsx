import { PayloadAction } from "@reduxjs/toolkit";
import { useDispatch } from "react-redux";
import { AppContainerStyled } from "./AppStyled";
import CardList from "./components/CardList/CardList";
import NewRobot from "./components/Form/Form";
import { User } from "./features/users/models/User";
import { loginUserAction } from "./features/users/slices/usersSlice";
import fetchToken from "./utils/auth";

const App = () => {
  const dispatch = useDispatch();

  const token = localStorage.getItem("token");
  if (token) {
    const user = fetchToken(token);
    dispatch<PayloadAction<User>>(loginUserAction(user));
  }

  return (
    <>
      <AppContainerStyled>
        <NewRobot />
        <CardList />
      </AppContainerStyled>
    </>
  );
};

export default App;
