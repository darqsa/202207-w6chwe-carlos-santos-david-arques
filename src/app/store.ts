import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import robotsSlice from "../features/robots/slices/robotsSlice";
import usersSlice from "../features/users/slices/usersSlice";

export const store = configureStore({
  reducer: {
    robots: robotsSlice,
    users: usersSlice,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
