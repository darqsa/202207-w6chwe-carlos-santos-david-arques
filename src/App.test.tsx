import { Provider } from "react-redux";
import TestRenderer from "react-test-renderer";
import App from "./App";
import { store } from "./app/store";

describe("Given an App component", () => {
  describe("When it's rendered and gets snapshotted", () => {
    test("Then it should show a show a snapshot of App", () => {
      const expectedAppSnapshot = TestRenderer.create(
        <Provider store={store}>
          <App />
        </Provider>
      );

      expect(expectedAppSnapshot).toMatchSnapshot();
    });
  });
});
