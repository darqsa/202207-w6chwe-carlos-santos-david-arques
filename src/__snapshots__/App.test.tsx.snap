// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When it's rendered and gets snapshotted Then it should show a show a snapshot of App 1`] = `
<div
  className="sc-bczRLJ iZsZhA"
>
  <div
    className="sc-hKMtZM bXbRhA form-container"
  >
    <h2
      className="form-container__title"
    >
      Create Your Robot
    </h2>
    <form
      autoComplete="off"
      className="form-container__form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-container__item-container"
      >
        <label
          className="form-container__item-label"
          htmlFor="name"
        >
          Name:
           
        </label>
        <input
          className="form-container__item"
          id="name"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-container__item-container"
      >
        <label
          className="form-container__item-label"
          htmlFor="image"
        >
          Image:
           
        </label>
        <input
          className="form-container__item"
          id="image"
          onChange={[Function]}
          required={true}
          type="url"
          value=""
        />
      </div>
      <div
        className="form-container__item-container"
      >
        <label
          className="form-container__item-label"
          htmlFor="speed"
        >
          Speed:
           
        </label>
        <input
          className="form-container__item"
          id="speed"
          max={10}
          min={1}
          onChange={[Function]}
          required={true}
          type="number"
          value={0}
        />
      </div>
      <div
        className="form-container__item-container"
      >
        <label
          className="form-container__item-label"
          htmlFor="image"
        >
          Strength:
           
        </label>
        <input
          className="form-container__item"
          id="strength"
          max={10}
          min={1}
          onChange={[Function]}
          required={true}
          type="number"
          value={0}
        />
      </div>
      <div
        className="form-container__item-container"
      >
        <label
          className="form-container__item-label"
          htmlFor="creationDate"
        >
          Creation Date:
           
        </label>
        <input
          className="form-container__item"
          id="creationDate"
          onChange={[Function]}
          type="date"
          value=""
        />
      </div>
      <button
        className="form-container__button"
        disabled={true}
        type="submit"
      >
        Create
      </button>
    </form>
  </div>
  <ul
    className="sc-dkzDqf eRMssT card-list"
  >
    <li
      className="card-list__item"
    >
      <div
        className="sc-gsnTZi ktktXK robot-container"
      >
        <div
          className="robot-container__image-container"
        >
          <img
            alt="It's just a robot named "
            className="robot-container__image"
            height={150}
            src=""
          />
        </div>
        <div
          className="robot-container__info-container"
        >
          <h2
            className="robot-container__name"
          />
          <ul
            className="robot-container__statistics-list"
          >
            <li
              className="robot-container__statistic-container"
            >
              <h3
                className="robot-container__statistic"
              >
                Speed: 
              </h3>
              <span
                className="robot-container__statistic-value"
              >
                1
              </span>
            </li>
            <li
              className="robot-container__statistic-container"
            >
              <h3
                className="robot-container__statistic"
              >
                Strength: 
              </h3>
              <span
                className="robot-container__statistic-value"
              >
                1
              </span>
            </li>
            <li
              className="robot-container__statistic-container"
            >
              <h3
                className="robot-container__statistic"
              >
                Creation Date: 
              </h3>
              <span
                className="robot-container__statistic-value"
              />
            </li>
          </ul>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
